debugger/src/debug_adapter.rs

```diff
@@ -1297,6 +1297,35 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         )
     }
 
+    /// Update the progress report in VSCode.
+    /// The progress has the range [0..1].
+    pub fn update_progress(
+        &mut self,
+        progress: f64,
+        message: Option<impl Into<String>>,
+        progress_id: i64,
+    ) -> Result<ProgressId> {
+        anyhow::ensure!(
+            self.supports_progress_reporting,
+            "Progress reporting is not supported by client."
+        );
+
+        let ok = self.send_event(
+            "progressUpdate",
+            Some(ProgressUpdateEventBody {
+                message: message.map(|v| v.into()),
+                percentage: Some(progress * 100.0),
+                progress_id: progress_id.to_string(),
+            }),
+        );
+
+        if ok.is_ok() {
+            Ok(progress_id)
+        } else {
+            Err(anyhow!("Failed to send event."))
+        }
```

Comment:
This could be simplified, by doing `self.send_event(...)?`, and then just returning `Ok(progress_id)`.